Braintree/src/main/java/com/braintreepayments/api/BraintreeFragment.java

```diff
@@ -135,8 +135,20 @@ public static BraintreeFragment newInstance(AppCompatActivity activity, String a
             throw new InvalidArgumentException("Activity is null");
         }
 
-        FragmentManager fm = activity.getSupportFragmentManager();
-        BraintreeFragment braintreeFragment = (BraintreeFragment) fm.findFragmentByTag(TAG);
+        return newInstance(activity, activity.getSupportFragmentManager(), authorization);
+    }
+
+    public static BraintreeFragment newInstance(Context context, FragmentManager fragmentManager, String authorization)
```

Comment:
What about a public method for passing a Fragment? 

This function would be a private method that both our public `newInstance` would delegate to.

Reply:
Haha. Actually, that was my initial idea. I thought you'd like this one as the public API. Making the change...

Reply:
Done.